From 62905c79684e3b4b38ed4f5e269f0bf9fcf25ed8 Mon Sep 17 00:00:00 2001
From: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
Date: Tue, 28 Sep 2021 17:16:51 +0300
Subject: [PATCH 071/233] [FIO extras] Kconfig: fix selecting FSL_CAAM without
 HAS_CAAM

The scheme SPL->OPTEE->U-Boot requires SPL to not use CAAM module
to avoid issues with a job ring in OP-TEE.
Now FSL_CAAM implicitly enables by several target boards even if
HAS_CAAM is disabled.
Fix selection of FSL_CAAM adding dependency from HAS_CAAM and
changing selection to non-mandatory.

Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>
---
 arch/arm/mach-imx/imx8m/Kconfig   | 32 +++++++++++++++----------------
 arch/arm/mach-imx/imx8ulp/Kconfig |  8 ++++----
 arch/arm/mach-imx/mx6/Kconfig     | 24 +++++++++++------------
 arch/arm/mach-imx/mx7/Kconfig     |  4 ++--
 arch/arm/mach-imx/mx7ulp/Kconfig  |  4 ++--
 cmd/Kconfig                       |  2 +-
 drivers/crypto/fsl/Kconfig        |  1 +
 8 files changed, 50 insertions(+), 49 deletions(-)

diff --git a/arch/arm/mach-imx/imx8m/Kconfig b/arch/arm/mach-imx/imx8m/Kconfig
index 9c28b737658..8e1766bc248 100644
--- a/arch/arm/mach-imx/imx8m/Kconfig
+++ b/arch/arm/mach-imx/imx8m/Kconfig
@@ -59,8 +59,8 @@ config TARGET_IMX8MQ_EVK
 	bool "imx8mq_evk"
 	select IMX8MQ
 	select IMX8M_LPDDR4
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 	select SPL_CRYPTO if SPL
 
@@ -96,8 +96,8 @@ config TARGET_IMX8MM_EVK
 	select IMX8MM
 	select SUPPORT_SPL
 	select IMX8M_LPDDR4
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 	select SPL_CRYPTO if SPL
 
@@ -106,8 +106,8 @@ config TARGET_IMX8MM_DDR4_EVK
 	select IMX8MM
 	select SUPPORT_SPL
 	select IMX8M_DDR4
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 	select SPL_CRYPTO if SPL
 
@@ -166,8 +166,8 @@ config TARGET_IMX8MN_EVK
 	select IMX8MN
 	select SUPPORT_SPL
 	select IMX8M_LPDDR4
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select SPL_CRYPTO if SPL
 
 config TARGET_IMX8MN_DDR4_EVK
@@ -175,8 +175,8 @@ config TARGET_IMX8MN_DDR4_EVK
 	select IMX8MN
 	select SUPPORT_SPL
 	select IMX8M_DDR4
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select SPL_CRYPTO if SPL
 
 config TARGET_IMX8MN_DDR3_EVK
@@ -184,8 +184,8 @@ config TARGET_IMX8MN_DDR3_EVK
 	select IMX8MN
 	select SUPPORT_SPL
 	select IMX8M_DDR3L
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select SPL_CRYPTO if SPL
 
 config TARGET_IMX8MN_AB2
@@ -227,8 +227,8 @@ config TARGET_IMX8MP_EVK
 	select IMX8MP
 	select SUPPORT_SPL
 	select IMX8M_LPDDR4
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 	select SPL_CRYPTO if SPL
 
@@ -237,8 +237,8 @@ config TARGET_IMX8MP_DDR4_EVK
 	select IMX8MP
 	select SUPPORT_SPL
 	select IMX8M_DDR4
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 	select SPL_CRYPTO if SPL
 
diff --git a/arch/arm/mach-imx/imx8ulp/Kconfig b/arch/arm/mach-imx/imx8ulp/Kconfig
index e82347ed687..7809ba70821 100644
--- a/arch/arm/mach-imx/imx8ulp/Kconfig
+++ b/arch/arm/mach-imx/imx8ulp/Kconfig
@@ -28,8 +28,8 @@ config TARGET_IMX8ULP_EVK
 	select IMX8ULP
 	select SUPPORT_SPL
 	select IMX8ULP_DRAM
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 	select SPL_CRYPTO if SPL
 
@@ -38,8 +38,8 @@ config TARGET_IMX8ULP_9X9_EVK
 	select IMX8ULP
 	select SUPPORT_SPL
 	select IMX8ULP_DRAM
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 	select SPL_CRYPTO_SUPPORT if SPL
 
diff --git a/arch/arm/mach-imx/mx6/Kconfig b/arch/arm/mach-imx/mx6/Kconfig
index 041b60c055d..d399e3fed3f 100644
--- a/arch/arm/mach-imx/mx6/Kconfig
+++ b/arch/arm/mach-imx/mx6/Kconfig
@@ -134,8 +134,8 @@ config TARGET_MX6SABREAUTO_COMMON
     select BOARD_EARLY_INIT_F
     select NXP_BOARD_REVISION
     imply CMD_DM
-    select FSL_CAAM
-    select FSL_BLOB
+    imply FSL_CAAM
+    imply FSL_BLOB
     select ARCH_MISC_INIT
     select RNG_SELF_TEST
 
@@ -148,8 +148,8 @@ config TARGET_MX6SABRESD_COMMON
     select BOARD_EARLY_INIT_F
     select NXP_BOARD_REVISION
     imply CMD_DM
-    select FSL_CAAM
-    select FSL_BLOB
+    imply FSL_CAAM
+    imply FSL_BLOB
     select ARCH_MISC_INIT
     select RNG_SELF_TEST
 
@@ -401,7 +401,7 @@ config TARGET_MX6SABREAUTO
 	select DM_THERMAL
 	select SUPPORT_SPL
 	imply CMD_DM
-	select FSL_CAAM
+	imply FSL_CAAM
 	select ARCH_MISC_INIT
 
 config TARGET_MX6SABRESD
@@ -413,7 +413,7 @@ config TARGET_MX6SABRESD
 	select DM_THERMAL
 	select SUPPORT_SPL
 	imply CMD_DM
-	select FSL_CAAM
+	imply FSL_CAAM
 	select ARCH_MISC_INIT
 
 config TARGET_MX6QSABREAUTO
@@ -485,8 +485,8 @@ config TARGET_MX6SXSABRESD
 	select DM
 	select DM_THERMAL
 	select SUPPORT_SPL
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 	select RNG_SELF_TEST
 
@@ -536,8 +536,8 @@ config TARGET_MX6UL_9X9_EVK
 	select IMX_MODULE_FUSE
 	select OF_SYSTEM_SETUP
 	imply CMD_DM
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 
 config TARGET_MX6UL_14X14_EVK
@@ -550,8 +550,8 @@ config TARGET_MX6UL_14X14_EVK
 	select IMX_MODULE_FUSE
 	select OF_SYSTEM_SETUP
 	imply CMD_DM
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 
 config TARGET_MX6UL_14X14_DDR3_VAL
diff --git a/arch/arm/mach-imx/mx7/Kconfig b/arch/arm/mach-imx/mx7/Kconfig
index e1278f497b3..9036927ba17 100644
--- a/arch/arm/mach-imx/mx7/Kconfig
+++ b/arch/arm/mach-imx/mx7/Kconfig
@@ -73,8 +73,8 @@ config TARGET_MX7DSABRESD
 	select DM_THERMAL
 	select MX7D
 	imply CMD_DM
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 
 config TARGET_MX7D_12X12_LPDDR3_VAL
 	bool "Support mx7d_12x12_lpddr3_val"
diff --git a/arch/arm/mach-imx/mx7ulp/Kconfig b/arch/arm/mach-imx/mx7ulp/Kconfig
index 319dcf6099f..306c6b15bad 100644
--- a/arch/arm/mach-imx/mx7ulp/Kconfig
+++ b/arch/arm/mach-imx/mx7ulp/Kconfig
@@ -56,8 +56,8 @@ config TARGET_MX7ULP_EVK
 	bool "Support mx7ulp EVK board"
 	select MX7ULP
 	select SYS_ARCH_TIMER
-	select FSL_CAAM
-	select FSL_BLOB
+	imply FSL_CAAM
+	imply FSL_BLOB
 	select ARCH_MISC_INIT
 
 endchoice
diff --git a/cmd/Kconfig b/cmd/Kconfig
index 5ae3816cadc..c95105d8218 100644
--- a/cmd/Kconfig
+++ b/cmd/Kconfig
@@ -2018,7 +2018,7 @@ config CMD_AES
 config CMD_BLOB
 	bool "Enable the 'blob' command"
 	select FSL_BLOB
-	depends on !MX6ULL && !MX6SLL && !MX6SL
+	depends on !MX6ULL && !MX6SLL && !MX6SL && HAS_CAAM
 	select IMX_HAB if ARCH_MX6 || ARCH_MX7 || ARCH_MX7ULP || ARCH_IMX8M
 	help
 	  This is used with the Freescale secure boot mechanism.
diff --git a/drivers/crypto/fsl/Kconfig b/drivers/crypto/fsl/Kconfig
index da5955e31de..24a917d1bb4 100644
--- a/drivers/crypto/fsl/Kconfig
+++ b/drivers/crypto/fsl/Kconfig
@@ -1,5 +1,6 @@
 config FSL_CAAM
 	bool "Freescale Crypto Driver Support"
+	depends on HAS_CAAM
 	select SHA_HW_ACCEL
 	# hw_sha1() under drivers/crypto, and needed with SHA_HW_ACCEL
 	select MISC if DM
-- 
2.25.1

